### PR TITLE
Blockchain: Add signature verification method

### PR DESCRIPTION
**Pull request by fishseabowl**
## Description

Fixes pyrsia/pyrsia#730

This PR implements the block::verify method and refactors the block signature structure.

## PR Checklist

- [x] I've read the [contributing guidelines](https://github.com/pyrsia/.github/blob/main/contributing.md).
- [x] I've read ["What is a Good PR?"](https://github.com/pyrsia/pyrsia/blob/main/docs/get_involved/good_pr.md)
- [x] I've included a good title and brief description along with how to review them.
- [x] I've linked any associated an [issue](https://github.com/pyrsia/pyrsia/issues).

### Code Contributions

- [x] I've built the code `cargo build --all-targets` successfully.
- [x] I've run the unit tests `cargo test --workspace` and everything passes.
- [x] I've made sure my rust toolchain is current `rustup update`.

### CODE REVIEW

File: src/blockchain/src/structures/block.rs [hunk 1]
```diff
@@ -52,17 +59,32 @@ impl Block {
         Self {
             header,
             transactions,
-            signature: Signature::new(&bincode::serialize(&header.hash()).unwrap(), signing_key),
+            signature: BlockSignature {
```
Comment (johanvos) [inline src/blockchain/src/structures/block.rs:62]: Renaming this to block_signature might avoid confusing.

File: src/blockchain/src/structures/block.rs [hunk 2]
```diff
@@ -25,14 +26,20 @@ use super::transaction::Transaction;
 use crate::crypto::hash_algorithm::HashDigest;
 use crate::signature::Signature;
 
-pub type BlockSignature = Signature;
+pub type PublicKey = [u8; 32];
+#[derive(Serialize, Deserialize, Debug, Clone, Eq, PartialEq, Decode, Encode, Hash)]
+pub struct BlockSignature {
+    signature: Signature,
+    #[codec(skip)]
```
Comment (efrisch) [inline src/blockchain/src/structures/block.rs:33]: Don't we need the public key in the future to validate the signature when we read the blocks?

Comment (fishseabowl) [reply src/blockchain/src/structures/block.rs:33]: It depends on the signature algorithm. For ed25519_dalek, we need the public key to verify the signature

Comment (sbtaylor15) [reply src/blockchain/src/structures/block.rs:33]: We are currently rolling out using ed25519 keys.

**Review by sbtaylor15 — APPROVED** [review]
/lgtm

**Review by efrisch — APPROVED** [review]
👍

**Pull request merged**